src/test/acceptance/test_upload_analyze_delete_firmware.py

```diff
@@ -3,102 +3,108 @@
-        default_plugins = [p for p in plugins if p != 'unpacker' and plugins[p][2]['default']]
-        optional_plugins = [p for p in plugins if not (plugins[p][1] or plugins[p][2])]
-        for mandatory_plugin in mandatory_plugins:
-            assert (
-                f'id="{mandatory_plugin}"'.encode() not in rv.data
-            ), f'mandatory plugin {mandatory_plugin} found erroneously'
-        for default_plugin in default_plugins:
-            assert (
-                f'value="{default_plugin}" checked'.encode() in rv.data
-            ), f'default plugin {default_plugin} erroneously unchecked or not found'
-        for optional_plugin in optional_plugins:
-            assert (
-                f'value="{optional_plugin}" unchecked'.encode() in rv.data
-            ), f'optional plugin {optional_plugin} erroneously checked or not found'
-
-    def _show_analysis_page(self, test_client):
-        db = FrontEndDbInterface()
-        assert db.exists(test_fw_a.uid), 'Error: Test firmware not found in DB!'
-        rv = test_client.get(f'/analysis/{test_fw_a.uid}')
-        assert test_fw_a.uid.encode() in rv.data
-        assert test_fw_a.name.encode() in rv.data
-        assert b'test_class' in rv.data
-        assert b'test_vendor' in rv.data
-        assert b'test_part' in rv.data
-        assert b'unknown' in rv.data
-        assert test_fw_a.file_name.encode() in rv.data, 'file name not found'
-        assert b'Admin</button>' in rv.data, 'admin options not shown with disabled auth'
-
-    def _check_ajax_file_tree_routes(self, test_client):
-        rv = test_client.get(f'/ajax_tree/{test_fw_a.uid}/{test_fw_a.uid}')
-        assert b'"children":' in rv.data
-        rv = test_client.get(f'/ajax_root/{test_fw_a.uid}/{test_fw_a.uid}')
-        assert b'"children":' in rv.data
-
-    def _check_ajax_on_demand_binary_load(self, test_client):
-        rv = test_client.get(
-            '/ajax_get_binary/text_plain/d558c9339cb967341d701e3184f863d3928973fccdc1d96042583730b5c7b76a_62'
-        )
-        assert b'test file' in rv.data
-
-    def _show_analysis_details_file_type(self, test_client):
-        rv = test_client.get(f'/analysis/{test_fw_a.uid}/file_type')
-        assert b'application/zip' in rv.data
-        assert b'Zip archive data' in rv.data
-        assert b'<pre><code>' not in rv.data, 'generic template used instead of specific template -> sync view error!'
-
-    def _show_home_page(self, test_client):
-        rv = test_client.get('/')
-        assert test_fw_a.uid.encode() in rv.data, 'test firmware not found under recent analysis on home page'
-
-    def _re_do_analysis_get(self, test_client):
-        rv = test_client.get(f'/admin/re-do_analysis/{test_fw_a.uid}')
+def _upload_firmware_get(test_client):
+    rv = test_client.get('/upload')
+    assert b'<h3 class="mb-3">Upload Firmware</h3>' in rv.data, 'upload page not displayed correctly'
+
+    intercom = InterComFrontEndBinding()
```

Comment:
Why not use the fixture `intercom_frontend_binding`?

Reply:
Because there is no such fixture? :sweat_smile: 
Maybe we should add one, though?